### PR TITLE
55509 - Set Up Role-Based Access Control

### PR DESCRIPTION
**Pull request by RobertNugent04**
In this branch, I set up role based access control for the database, creating the admin & engineer users and assigning the appropriate permissions. 

**Roles and Permissions Created:**
- Admin - All Permssions
- Engineer:
  - SELECT on Machines Table
  - SELECT on Update_Logs Table
  - SELECT, INSERT, UPDATE on Packages Table
  
**RBAC Implementation:**
I am automatically assigning roles during user creation through a stored procedure, based on the role specified for the user in the Users table. This stored procedure would be used whenever a new user is being created for the system.

**CreateUser Stored Procedure Snippet:**
stored procedure

**Database Schema (for reference):**
final_final_final

### CODE REVIEW

**Review by kevashans — DISMISSED** [review]

File: upflux_db.sql [hunk 1]
```diff
@@ -3,125 +3,154 @@
...
+CREATE TABLE Licenses (
+    licence_key VARCHAR(255) NOT NULL PRIMARY KEY,
     machine_id INT NOT NULL,
     validity_status VARCHAR(50) NOT NULL,
     expiration_date TIMESTAMP NOT NULL,
-    PRIMARY KEY (licence_key),
     FOREIGN KEY (machine_id) REFERENCES Machines(machine_id)
 );
 
+-- Create Credentials Table
```
Comment (PatrickOrjieh) [inline upflux_db.sql:40]: Like we discussed, there might be a conflict here, when a token expores and that same user gets another token with the same access to the machines. So a better PK has to be used

Comment (RobertNugent04) [reply upflux_db.sql:40]: Thanks for the feedback. I have now added a new unique primary key for the credentials table as well as an 'expires_at' field to track when a token expires.

File: upflux_db.sql [hunk 2]
```diff
@@ -3,125 +3,156 @@
...
+SELECT * FROM Licenses;
 
-INSERT INTO Credentials (machine_id, user_id, access_level, access_granted_at) VALUES
-(1, 1, 'Admin', '2023-01-01 12:00:00'),
-(2, 2, 'Write', '2023-03-15 09:30:00'),
-(3, 3, 'Read', '2023-05-20 11:45:00');
+/* Inserting data into Credentials */
+INSERT INTO Credentials (user_id, machine_id, access_granted_at) VALUES
+(1, 1, '2023-01-01 12:00:00'),
+(2, 2, '2023-03-15 09:30:00'),
```
Comment (PatrickOrjieh) [inline upflux_db.sql]: Update to reflect the change

**Review by PatrickOrjieh — APPROVED** [review]
Great Job

**Review by kevashans — APPROVED** [review]

**Pull request merged**